### PR TITLE
feat: update guide for discord.js v14 and sapphire v4

### PR DESCRIPTION
**Pull request by favna**
Related:
- [framework#512](https://github.com/sapphiredev/framework/pull/512)
- [utilities#434](https://github.com/sapphiredev/utilities/pull/434)
- [plugins#403](https://github.com/sapphiredev/plugins/pull/403)

This PR brings updates the guide for djs v14 and Sapphire v4.

### CODE REVIEW

File: docs/General/Welcome.mdx [hunk 1]
```diff
@@ -54,7 +54,7 @@ become the ultimate modern experience of writing your code.
 **Node.js 16.6.0 or newer is required.**
 
 ```bash npm2yarn2pnpm
-npm install @sapphire/framework discord.js@13.x
+npm install @sapphire/framework discord.js@latest
```
Comment (vladfrangu) [inline docs/General/Welcome.mdx:57]: Is there a 14.x tag? Or better yet, we should probably do it like 14.x too

File: docs/Guide/getting-started/getting-started-with-sapphire.mdx [hunk 2]
```diff
@@ -3,11 +3,11 @@ sidebar_position: 0
...
 ---
 
-To install Sapphire, you need to install both `discord.js` and `@sapphire/framework`. You can do this by running the
-following command:
+To install Sapphire, you need to install both [`discord.js`](https://discord.js.org) and
+[`@sapphire/framework`](../../../). You can do this by running the following command:
 
 ```bash npm2yarn2pnpm
-npm install @sapphire/framework discord.js@13.x
+npm install @sapphire/framework discord.js@latest
```
Comment (vladfrangu) [inline docs/Guide/getting-started/getting-started-with-sapphire.mdx:10]: Ditto

File: docs/Guide/getting-started/updating-from-v3-to-v4.mdx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+## Before you start
+
+It is important to note that the primary breaking change of Sapphire v4 is support for DiscordJS v14.x. This means that
+for the bulk of the breaking changes we want to refer you to [Updating from v13 to v14][djs-migration].
+
+## Sapphire specific changes
+
+Sapphire has a few breaking changes that are not related to DiscordJS. These are:
+
+- The [`Identifiers`][Identifiers] enum was changed from `const enum` to [`enum`][enum].
```
Comment (vladfrangu) [inline docs/Guide/getting-started/updating-from-v3-to-v4.mdx:16]: I wouldn't call the const enum -> enum changes breaking tbh, for users this doesn't really change anything (for TS users its.. insignificant)

File: docs/Guide/getting-started/updating-from-v3-to-v4.mdx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+- The [`CommandOptionsRunTypeEnum`][CommandOptionsRunTypeEnum] enum was changed from `const enum` to [`enum`][enum].
+- The [`CommandPreConditions`][CommandPreConditions] enum was changed from `const enum` to [`enum`][enum].
+- The [`InteractionHandlerTypes`][InteractionHandlerTypes] enum was changed from `const enum` to [`enum`][enum].
+- The [`InternalRegistryAPIType`][InternalRegistryAPIType] enum was changed from `const enum` to [`enum`][enum].
```
Comment (vladfrangu) [inline docs/Guide/getting-started/updating-from-v3-to-v4.mdx:20]: I wouldn't log this as its not exposed to users
And also iirc should still be const

File: docs/Guide/getting-started/updating-from-v3-to-v4.mdx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+- The [`LogLevel`][LogLevel] enum was changed from `const enum` to [`enum`][enum].
+- The [`PreconditionRunMode`][PreconditionRunMode] enum was changed from `const enum` to [`enum`][enum].
+- The type [`Command.ChatInputInteraction`][ChatInputInteraction] enum was changed to
```
Comment (vladfrangu) [inline docs/Guide/getting-started/updating-from-v3-to-v4.mdx:23]: Its an alias, not an enum

File: docs/Guide/plugins/API/getting-started.mdx [hunk 6]
```diff
@@ -15,7 +15,7 @@ to change those settings, and your website needs to interface with the bot for t
 ## Installation
 
 ```bash npm2yarn2pnpm
-npm install @sapphire/plugin-api @sapphire/framework discord.js@13.x discord-api-types@^0.33.5
+npm install @sapphire/plugin-api @sapphire/framework discord.js@latest discord-api-types@latest
```
Comment (vladfrangu) [inline docs/Guide/plugins/API/getting-started.mdx:18]: Ditto

File: docs/Guide/plugins/API/getting-started.mdx [hunk 7]
```diff
@@ -67,7 +67,7 @@ extension class.
 	// The URL that users should be redirected to after a successful authentication
     redirect: '',
 	// The scopes that should be given to the authentication
-    scopes: ['identify'],
+    scopes: [OAuth2Scopes.Identify],
```
Comment (vladfrangu) [inline docs/Guide/plugins/API/getting-started.mdx:70]: Should add the import to the example so users know where to get it from

File: docs/Guide/plugins/Subcommands/getting-started.mdx [hunk 8]
```diff
@@ -14,7 +14,7 @@ Groups][discord-subcommand-application-command-docs] interface for Chat Input Co
 ## Installation
 
 ```bash npm2yarn2pnpm
-npm install @sapphire/plugin-subcommands @sapphire/framework @sapphire/utilities discord.js@13.x
+npm install @sapphire/plugin-subcommands @sapphire/framework @sapphire/utilities discord.js@latest
```
Comment (vladfrangu) [inline docs/Guide/plugins/Subcommands/getting-started.mdx:17]: Ditto

File: docs/Guide/plugins/i18next/getting-started.mdx [hunk 9]
```diff
@@ -23,7 +23,7 @@ In short: **Everything that is possible with [i18next] is also possible with [`@
 ## Installation
 
 ```bash npm2yarn2pnpm
-npm install @sapphire/plugin-i18next @sapphire/framework discord.js@13.x
+npm install @sapphire/plugin-i18next @sapphire/framework discord.js@latest
```
Comment (vladfrangu) [inline docs/Guide/plugins/i18next/getting-started.mdx:26]: Ditto

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: docs/Guide/getting-started/updating-from-v3-to-v4.mdx [hunk 10]
```diff
@@ -0,0 +1,44 @@
+---
+sidebar_position: 5
+title: Updating from v3 to v4
+toc_max_heading_level: 5
+---
+
+## Before you start
+
+It is important to note that the primary breaking change of Sapphire v4 is support for DiscordJS v14.x. This means that
+for the bulk of the breaking changes we want to refer you to [Updating from v13 to v14][djs-migration].
```
Comment (Rexogamer) [inline docs/Guide/getting-started/updating-from-v3-to-v4.mdx:10]: This seems a bit clunky; maybe something like this?
```suggestion
for most of the breaking changes you should refer to [DiscordJS' upgrade guide][djs-migration].
```

**Pull request merged**